Building issues on multiple platforms

I think current version of H5 is not usable much.

1) `UpdateH5` option simply does not work as intended. If you have NET 7 installed it will update your installed h5 tools in any case. If you have only NET 6 then it cannot update and will not install new version. But enabling this option can reduce amount of error in the log.

2) It always puts a lot of different warnings which should not be displayed in case of normal building process:

For example:
`Warning	MSB3073	The command "dotnet tool run h5 check-if-online" exited with code 1`

![image](https://user-images.githubusercontent.com/4306334/202947694-f3954f77-4c25-4eee-aa69-f1e5ca1d3ef0.png)


Exit code other than zero means that external program completed not successful.


@theolivenbaum Can you provide any feedback? We cannot hear anything from you.

> Interesting, which version of VS do you have installed?

I guess it does not matter, because it happens even under gitlab-runner (CentOS 7, .NET 7, Microsoft Official Repository)

```
 h5 check-if-online
/home/gitlab-runner/.nuget/packages/h5.target/0.0.34005/Sdk/Sdk.targets(461,5): warning MSB3073: The command "h5 check-if-online" exited with code 134.
```

Difference only in exit code for different platforms, for VS and Windows it is 1 and for Linux it is any none-zero as I see that.

--- a/.devops/build-dotnet-tool.yml
+++ b/.devops/build-dotnet-tool.yml
@@ -2,6 +2,7 @@ variables:
   compiler:  '$(Build.SourcesDirectory)/H5/Compiler/Compiler/H5.Compiler.csproj'
   compilerService: '$(Build.SourcesDirectory)/H5/Compiler/Compiler.Service/H5.Compiler.Service.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -20,12 +21,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: DotNetCoreCLI@2
   displayName: 'restore nuget compiler'
@@ -44,14 +59,14 @@ steps:
   inputs:
     command: 'build'
     projects: '$(compiler)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'build compiler-as-a-service'
   inputs:
     command: 'build'
     projects: '$(compilerService)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget compiler'
@@ -60,7 +75,7 @@ steps:
     configuration: '$(buildConfiguration)'
     packagesToPack: '$(compiler)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget compiler-as-a-service'
@@ -69,13 +84,13 @@ steps:
     configuration: '$(buildConfiguration)'
     packagesToPack: '$(compilerService)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5-compiler.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5-compiler.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'
 
@@ -83,6 +98,6 @@ steps:
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.Compiler.Service.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.Compiler.Service.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-core.yml
+++ b/.devops/build-h5-core.yml
@@ -3,6 +3,7 @@ variables:
   h5core: '$(Build.SourcesDirectory)/H5/H5.Core/H5.Core.csproj'
   h5json: '$(Build.SourcesDirectory)/H5/H5.Newtonsoft.Json/H5.Newtonsoft.Json.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 
 
@@ -23,12 +24,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -46,7 +61,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5core)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -55,12 +70,12 @@ steps:
     packagesToPack: '$(h5core)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.core.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.core.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-howler.yml
+++ b/.devops/build-h5-howler.yml
@@ -1,6 +1,7 @@
 variables:
   h5howler: '$(Build.SourcesDirectory)/H5/H5.Packages/H5.howler/H5.howler.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,12 +20,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -42,7 +57,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5howler)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -51,12 +66,12 @@ steps:
     packagesToPack: '$(h5howler)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.howler.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.howler.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-httpclient.yml
+++ b/.devops/build-h5-httpclient.yml
@@ -1,6 +1,7 @@
 variables:
   h5httpclient: '$(Build.SourcesDirectory)/H5/H5.Packages/H5.httpclient/H5.httpclient.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,12 +20,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -42,7 +57,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5httpclient)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -51,12 +66,12 @@ steps:
     packagesToPack: '$(h5httpclient)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.httpclient.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.httpclient.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-json.yml
+++ b/.devops/build-h5-json.yml
@@ -1,6 +1,7 @@
 variables:
   h5json: '$(Build.SourcesDirectory)/H5/H5.Newtonsoft.Json/H5.Newtonsoft.Json.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,6 +20,9 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
@@ -26,6 +30,17 @@ steps:
     version: 7.x
     includePreviewVersions: true
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -43,7 +58,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5json)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -52,12 +67,12 @@ steps:
     packagesToPack: '$(h5json)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.Newtonsoft.Json.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.Newtonsoft.Json.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-lib.yml
+++ b/.devops/build-h5-lib.yml
@@ -3,6 +3,7 @@ variables:
   h5core: '$(Build.SourcesDirectory)/H5/H5.Core/H5.Core.csproj'
   h5json: '$(Build.SourcesDirectory)/H5/H5.Newtonsoft.Json/H5.Newtonsoft.Json.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 
 
@@ -22,6 +23,9 @@ pr: none
 
 steps:
 - checkout: self
+
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
   
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
@@ -29,6 +33,17 @@ steps:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -46,7 +61,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5base)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -55,12 +70,12 @@ steps:
     packagesToPack: '$(h5base)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-p2.yml
+++ b/.devops/build-h5-p2.yml
@@ -1,6 +1,7 @@
 variables:
   h5p2: '$(Build.SourcesDirectory)/H5/H5.Packages/H5.p2/H5.p2.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,12 +20,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -42,7 +57,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5p2)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -51,12 +66,12 @@ steps:
     packagesToPack: '$(h5p2)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.p2.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.p2.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-target.yml
+++ b/.devops/build-h5-target.yml
@@ -1,6 +1,7 @@
 variables:
   targetPath:  '$(Build.SourcesDirectory)/H5/H5.Build.Target/H5.Build.Target.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,12 +20,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: DotNetCoreCLI@2
   displayName: 'restore nuget'
@@ -39,12 +54,12 @@ steps:
     packagesToPack: '$(targetPath)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)"'
+    buildProperties: 'Version="$(targetVersion)"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '$(Build.ArtifactStagingDirectory)/h5.Target.0.0.$(build.buildId).nupkg'
+    packagesToPush: '$(Build.ArtifactStagingDirectory)/h5.Target.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-h5-webgl2.yml
+++ b/.devops/build-h5-webgl2.yml
@@ -1,6 +1,7 @@
 variables:
   h5webgl2: '$(Build.SourcesDirectory)/H5/H5.Packages/H5.webgl2/H5.webgl2.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -19,12 +20,26 @@ pr: none
 steps:
 - checkout: self
   
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
+
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
   inputs:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: CmdLine@2
   displayName: 'Install h5 compiler'
@@ -42,7 +57,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(h5webgl2)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)  /p:AllowUnsafeBlocks=True /p:LangVersion=latest'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -51,12 +66,12 @@ steps:
     packagesToPack: '$(h5webgl2)'
     configuration: '$(buildConfiguration)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)";AllowUnsafeBlocks="True";LangVersion="latest"'
+    buildProperties: 'Version="$(targetVersion)";AllowUnsafeBlocks="True";LangVersion="latest"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '**/h5.webgl2.0.0.$(build.buildId).nupkg'
+    packagesToPush: '**/h5.webgl2.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/.devops/build-template.yml
+++ b/.devops/build-template.yml
@@ -1,6 +1,7 @@
 variables:
   project: '$(Build.SourcesDirectory)/Template/h5.Template.csproj'
   buildConfiguration: 'Release'
+  targetVersion: yy.M.$(build.buildId)
 
 pool:
   vmImage: 'windows-latest'
@@ -18,6 +19,9 @@ pr: none
 
 steps:
 - checkout: self
+
+- script: |
+    echo "Commit message: $(Build.SourceVersionMessage)"
   
 - task: UseDotNet@2
   displayName: 'Use .NET 7.0 SDK'
@@ -25,6 +29,17 @@ steps:
     packageType: sdk
     version: 7.x
     installationPath: $(Agent.ToolsDirectory)/dotnet
+
+- task: PowerShell@2
+  displayName: 'Create CalVer Version'
+  inputs:
+    targetType: 'inline'
+    script: |
+      $dottedDate = (Get-Date).ToString("yy.M")
+      $buildID = $($env:BUILD_BUILDID) 
+      $newTargetVersion = "$dottedDate.$buildID"
+      Write-Host "##vso[task.setvariable variable=targetVersion;]$newTargetVersion"
+      Write-Host "Updated targetVersion to '$newTargetVersion'"
 
 - task: DotNetCoreCLI@2
   displayName: 'restore nuget'
@@ -37,7 +52,7 @@ steps:
   inputs:
     command: 'build'
     projects: '$(project)'
-    arguments: '-c $(buildConfiguration) /p:Version=0.0.$(build.buildId)'
+    arguments: '-c $(buildConfiguration) /p:Version=$(targetVersion)'
 
 - task: DotNetCoreCLI@2
   displayName: 'pack nuget'
@@ -46,12 +61,12 @@ steps:
     configuration: '$(buildConfiguration)'
     packagesToPack: '$(project)'
     versioningScheme: 'off'
-    buildProperties: 'Version="0.0.$(build.buildId)"'
+    buildProperties: 'Version="$(targetVersion)"'
 
 - task: NuGetCommand@2
   displayName: 'push nuget'
   inputs:
     command: 'push'
-    packagesToPush: '$(Build.ArtifactStagingDirectory)/h5.Template.0.0.$(build.buildId).nupkg'
+    packagesToPush: '$(Build.ArtifactStagingDirectory)/h5.Template.$(targetVersion).nupkg'
     nuGetFeedType: 'external'
     publishFeedCredentials: 'nuget-curiosity'

--- a/H5/H5.Build.Target/Sdk/Sdk.targets
+++ b/H5/H5.Build.Target/Sdk/Sdk.targets
@@ -458,21 +458,21 @@
   <Target Name="_CheckForH5">
     <!-- If the compiler is not running, or if not found, set the flag to update it -->
 
-    <Exec Command="h5 check-if-online" ConsoleToMsBuild="false" WorkingDirectory="$(ProjectDir)" ContinueOnError="true" IgnoreExitCode="false">
+    <Exec Command="h5 check-if-online" ConsoleToMsBuild="false" WorkingDirectory="$(ProjectDir)" ContinueOnError="true" IgnoreExitCode="true">
       <Output TaskParameter="ExitCode" PropertyName="ExitCode" />
     </Exec>
 
-    <PropertyGroup  Condition="'$(MSBuildLastTaskResult)' == 'False' OR '$(ExitCode)' == '1'">
+    <PropertyGroup  Condition="'$(ExitCode)' == '1'">
       <UpdateH5>true</UpdateH5>
     </PropertyGroup>
 
     <!--Check for local compiler-->
-    <Exec Command="dotnet tool run h5 check-if-online" ConsoleToMsBuild="false" WorkingDirectory="$(ProjectDir)" ContinueOnError="true" IgnoreExitCode="false" Condition="!$(UseLocalH5)">
+    <Exec Command="dotnet tool run h5 check-if-online" ConsoleToMsBuild="false" WorkingDirectory="$(ProjectDir)" ContinueOnError="true" IgnoreExitCode="true" Condition="!$(UseLocalH5)">
       <Output TaskParameter="ExitCode" PropertyName="ExitCodeLocal" />
     </Exec>
     
     <PropertyGroup  Condition="'UseLocalH5'=='true' OR '$(ExitCodeLocal)' == '0'">
-      <UpdateH5>false</UpdateH5>
+      <UpdateH5>false</UpdateH5> <!-- We don't update the local install -->
       <UseLocalH5>true</UseLocalH5>
     </PropertyGroup>
 
@@ -488,7 +488,7 @@
 
     <Message Importance="high" Text="Installing (or updating) the h5 compiler." Condition="$(UpdateH5)" />
     <Message Importance="high" Text="Skipping updating the h5 compiler. You can always install/update it by calling dotnet tool update --global h5-compiler" Condition="!$(UpdateH5) AND !$(UseLocalH5)" />
-    <Exec Command="dotnet tool update --global h5-compiler" Condition="$(UpdateH5)"  ContinueOnError="WarnAndContinue"/>
+    <Exec Command="dotnet tool update --global h5-compiler" Condition="$(UpdateH5)" ContinueOnError="WarnAndContinue"/>
 
     <Message Importance="high" Text="Deleting previous assembly: $(ProjectDir)$(OutputPath)$(TargetName)$(TargetExt)"/>
     <Delete Files="$(ProjectDir)$(OutputPath)$(TargetName)$(TargetExt)"/>

